sklearn/decomposition/_lda.py

```diff
@@ -587,10 +594,12 @@ def fit(self, X, y=None):
             Document word matrix.
 
         y : Ignored
+            Not used, present here for API consistency by convention.
 
         Returns
         -------
-        self
+        self:
```

Comment:
```suggestion
        self
```

Reply:
`self : object`

Reply:
The initial version was correct  for cases where there are no types, and it's a bit pointless to indicate `object` dtype for estimators https://numpydoc.readthedocs.io/en/latest/format.html#parameters 

cc @glemaitre

Reply:
> it's a bit pointless to indicate object dtype for estimators which is also not accurate

I agree. However, we have this way of documenting everywhere. I would prefer to keep this inaccurate convention until we do a find/replace regex in another PR. Would it not be easier to make the change?

Reply:
Right, but my point is there is no issue with this line of docstring. It should pass the numpydoc validation, I think? And if so I would rather we didn't change it instead of introducing unnecessary code churn with a solution we know is not helpful / correct.

`self : object` is used elsewhere, but 27% of estimators use the current conversion,
```bash
$ rg "^\s+self$"  | grep self | wc -l
38
$ rg "^\s+self\s*:$"  | grep self | wc -l
2
$ rg "^\s+self\s*:\s*object$"  | grep self | wc -l
103
```

Reply:
@rth Thank you so much for reviewing my PR. The numpydoc validation requires a description and that's the reason a added a tautology. I'll be more than happy to make any changes according to your guidance.

Reply:
I think the description is ok. The point of @rth is about `self : object` vs `self` alone on the current line, not the description below.